### PR TITLE
bugfix/86cvc73ph: Fixed service page bug with not loading

### PR DESCRIPTION
**Pull request by NikkiShao**
Added react Loader (all different types viewable at the bottom of the example page)

Added loader for service page when loading data, this solves the bug of items not appearing when moving to the page from another page.

### CODE REVIEW

File: behind-the-veil-siteroot/imports/ui/components/pages/service/ServicesPage.jsx [hunk 1]
```diff
@@ -63,18 +66,25 @@ export const ServicesPage = () => {
...
+        serviceId={service._id._str}
+        serviceName={service.serviceName}
+        serviceDesc={service.serviceDesc}
+        servicePrice={service.servicePrice}
+        artistUsername={service.artistUsername}
+        serviceImageData={service.serviceImageData}
+        artistAlias={service.artistAlias}
+    ></ServiceCard>))
+
+    const [isLoading, setIsLoading] = useState(true);
```
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/pages/service/ServicesPage.jsx:79]: i think isLoading shouldnt be a state either. looking at the useSubscribe docs, when you call one of those isLoading functions, react will rerender whenever a loading state is updated, so you can replace the state and the useeffect with
` const isLoading = isLoadingService() || isLoadingArtists() || isLoadingImages() ` 
to get rid of unnecessary rerenders.

File: behind-the-veil-siteroot/imports/ui/components/pages/examples/loaderExamples/LoaderExamples.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ */
+import React from "react";
+
+/**
+ * General loader component.
+ */
+const LoaderExamples = () => {
+    return (<div className="flex flex-col gap-3">
+            <div className="large-text underline">Example Loaders (we should pick one/two):</div>
+            <div className={"flex flex-row items-center justify-center flex-wrap gap-x-48 gap-y-32"}>
```
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/pages/examples/loaderExamples/LoaderExamples.jsx:36]: could also include the Loader component you created in these examples too

**Review by joshualoongwy — CHANGES_REQUESTED** [review]
couple state related things

File: behind-the-veil-siteroot/imports/ui/components/pages/examples/loaderExamples/LoaderExamples.jsx (hunk 2 above)
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/pages/examples/loaderExamples/LoaderExamples.jsx:36]: oki

File: behind-the-veil-siteroot/imports/ui/components/loader/Loader.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {number} size - size of the loader
+ * @param {number} speed - speed of the loader's animation
+ * @param loaderProps - other properties to pass into the loader
+ */
+const Loader = ({className, loadingText, isLoading, size, speed,...loaderProps}) => {
+
+    const loaderClasses = classNames("flex flex-col items-center justify-center gap-y-10", className);
+
+    // text that is displayed under the moving loading icon
+    const [text, setText] = React.useState(loadingText);
```
[the remark this answers is not in the document]
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/loader/Loader.jsx:26]: oo nicee! will do

File: behind-the-veil-siteroot/imports/ui/components/pages/service/ServicesPage.jsx (hunk 1 above)
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/pages/service/ServicesPage.jsx]: changed and removed the useeffect! thank you!

**Review by joshualoongwy — APPROVED** [review]

**Pull request merged**